### DISCUSSION TITLE
need help in PHP

### DISCUSSION THREAD

**Discussion by Mackie22w** [EddieHubCommunity/support#755] [Q&A]
<code> <?php
// This is to check if the request is coming from a specific domain 
$ref = $_SERVER['HTTP_REFERER'];
$refData = parse_url($ref);

if($refData['host'] !== 'domain') {
  // Output string and stop execution
  die("Hotlinking not permitted");
}

echo "Executing code here";
?> <code>

Please tell how to add multiple domains in this execution

I also tried-     if($refData['host'] != 'website1.com' || $refData['host'] != 'website2.com') {

**Comment by eddiejaoude** [comment]
From your code ...

```php
<?php
// This is to check if the request is coming from a specific domain 
$ref = $_SERVER['HTTP_REFERER'];
$refData = parse_url($ref);

if($refData['host'] !== 'domain') {
  // Output string and stop execution
  die("Hotlinking not permitted");
}

echo "Executing code here";
```

Q. When you use the OR condition in the `if` can you share a screenshot of the error?

Also you could put the options in an array to allow it to be scalable and cleaner, for example

> in_array ( mixed $needle , array $haystack , bool $strict = false ) : bool

From the [php docs](https://www.php.net/manual/en/function.in-array.php)

```php
if (in_array($refData['host'], [ 'domain1', 'domain2', 'domain3' ])) {
```

**Comment by Mackie22w** [comment]
actually, this was the real code I was trying to use

```
<?php
// This is to check if the request is coming from a specific domain
$ref = $_SERVER['HTTP_REFERER'];
$refData = parse_url($ref);

if($refData['host'] !== 'ecopeak.net') {
  // Output string and stop execution
  die("<meta http-equiv='refresh' content='1;url=http://slayygames.com/' /> Not Allowed.");
}

?>
```

but when I try to put other code (Multiple domains)  it doesn't work

this code work when I use single domain

**Reply by stemount** [reply]
I think @eddiejaoude has answered your question here @Mackie22w but if you look at his example

```php
if (in_array($refData['host'], [ 'domain1', 'domain2', 'domain3' ])) {
```

what you actually want is the opposite of `in_array` being true (e.g. you want to whitelist domains that can use this)

try this:

```php
if (!in_array($refData['host'], [ 'domain1', 'domain2', 'domain3' ])) {
  // Output string and stop execution
  die("<meta http-equiv='refresh' content='1;url=http://slayygames.com/' /> Not Allowed.");
}
```
